### ISSUE TITLE
Fatal error: Call to undefined method MongoDB\Driver\ReadConcern::isDefault() in ..src\Operation\Find.php on line 193

### ISSUE THREAD

**Issue by AlexGnatko** [mongodb/mongo-php-library#412]
The code used to work with an older version of mongo-php-library, but now it doesn't.
I provide empty $options when I set up the connection.

**Comment by AlexGnatko** [comment]
It's interesting, but now I realize that MongoDB\Driver\ReadConcern::isDefault() doesn't work at all, even on good ReadConcern objects. getLevel() works, but isDefault() doesn't... Am I using some wrong MongoDB driver?

**Comment by AlexGnatko** [comment]
Oh, sure, I'm using 1.2.3 (that's what "apt install php-mongodb" gave me) and isDefault() appeared only in 1.3. OK, closing.

**Comment by jmikola** [comment]
Note that the `master` branch of the library (what will become 1.2.0) requires version 1.3.0+ of the extension (https://github.com/mongodb/mongo-php-library/commit/6e091545d42d194116436f787c0692be94cc870b). Installing the library via Composer and allowing it to check platform dependencies (default behavior unless suppressed with `--ignore-platform-reqs`) would have caught this immediately.

**Comment by kamihouse** [comment]
Hi guys;
Apparently this is a method that was taken from later version of the library `mongodb/mongodb`.

This method is still available in the version: `1.1.2`.

To resolve this issue temporarily, update the `composer.lock` file by replacing the following content:

```
    {
            "name": "mongodb/mongodb",
            "version": "1.1.2",
            "source": {
                "type": "git",
                "url": "https://github.com/mongodb/mongo-php-library.git",
                "reference": "a307dd60e71e1291c60927ea4f3a1905146063f5"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/mongodb/mongo-php-library/zipball/a307dd60e71e1291c60927ea4f3a1905146063f5",
                "reference": "a307dd60e71e1291c60927ea4f3a1905146063f5",
                "shasum": ""
            },
            "require": {
                "ext-mongodb": "^1.2.0",
                "php": ">=5.4"
            },
            "require-dev": {
                "phpunit/phpunit": "^4.8"
            },
            "type": "library",
            "autoload": {
                "psr-4": {
                    "MongoDB\\": "src/"
                },
                "files": [
                    "src/functions.php"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "Apache-2.0"
            ],
            "authors": [
                {
                    "name": "Jeremy Mikola",
                    "email": "jmikola@gmail.com"
                },
                {
                    "name": "Hannes Magnusson",
                    "email": "bjori@mongodb.com"
                },
                {
                    "name": "Derick Rethans",
                    "email": "github@derickrethans.nl"
                }
            ],
            "description": "MongoDB driver library",
            "homepage": "https://jira.mongodb.org/browse/PHPLIB",
            "keywords": [
                "database",
                "driver",
                "mongodb",
                "persistence"
            ],
            "time": "2017-02-16T18:40:32+00:00"
        },
```

**Comment by jmikola** [comment]
> Apparently this is a method that was taken from latest version of the library mongodb/mongodb.
>
> This method is still available in the version: 1.1.2.

I'm not sure I understand your last comment. The `isDefault()` method has no relation to library version 1.1.x.

The library's call to `MongoDB\Driver\ReadConcern::isDefault()` was added in https://github.com/mongodb/mongo-php-library/commit/236657b5ce894fa811200369c93a1c29f761db22, which was only added in library version 1.2.0. That version of the library also [depends on `ext-mongodb` 1.3.0+](https://github.com/mongodb/mongo-php-library/blob/1.2.0/composer.json#L16), which is the version of the extension in which `MongoDB\Driver\ReadConcern::isDefault()` first appears.
